metricflow/engine/models.py

```diff
@@ -1,19 +1,35 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import List
+from typing import List, Optional
+
+from dbt_semantic_interfaces.objects.metric import MetricType, MetricTypeParams, Metric as PydanticMetric
```

Comment:
Might be worth calling PydanticMetric something like SemanticManifestMetric or similar. It's so long though. ModelMetric would've been nice. Sigh.

Reply:
I don't mind long!